include/deal.II/base/mpi.templates.h

```diff
@@ -264,6 +264,26 @@ namespace Utilities
 
 
 
+    template <typename Number>
+    void
+    sum (const SparseMatrix<Number> &local,
```

Comment:
@bangerth i hope you find this solution to your satisfaction. I will remove the `make_array_view()` once we reach consensus.

Reply:
this approach addresses 

> By creating the ArrayView
constructor, you make it impossible to ever change the underlying data
structure again in a backward-compatible way

Reply:
> I will remove the make_array_view() 

done.